src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part1/index.md

```diff
@@ -0,0 +1,430 @@
+
+## Overview
+
+For each scheduler, the tests involved running client ops on a RBD pool with the following background operations:
+
+- Test A: Snaptrim ops on a separate pool
+- Test B: Background recovery and snaptrim ops in parallel on separate pools.
+
+The results were collated and the following statistics were compared between the two schedulers for the following operation types:
+
+- External client
+    - Average throughput (IOPS)
+    - Average completion and percentile (95th, 99th, 99.5th) latencies
+- Background recovery
+    - Average recovery throughput
+    - Number of misplaced objects recovered per second
+- Background snaptrim
+    - Average snaptrim rate
+    - Number of objects trimmed
+
+## Test Environment
+
+A single node with the following configuration was used for the tests:
+
+- **Software Configuration**: CentOS 8.1.1911 Linux Kernel 4.18.0-193.6.3.el8_2.x86_64
+- **CPU**: 2 x Intel® Xeon® CPU E5-2650 v3 @ 2.30GHz
+- **nproc**: 40
+- **System Memory**: 64 GiB
+- **Tuned-adm Profile**: network-latency
+- **Ceph Version**: 17.0.0-12483-g307e20ec647 (307e20ec64724620831e1c3f0ad806562c80592a) quincy (dev)
+- **Storage**: Intel® NVMe SSD DC P3700 Series (SSDPE2MD800G4) [4 x 800GB]
+
+All the Ceph pools were configured with a replication factor of 3. A total of 4 OSDs were configured on the node for the tests.
+
+## Test Methodology
+
+Ceph [cbt](https://github.com/ceph/cbt) was used to test the scenarios. A new test to generate background snaptrim ops and recovery ops with client I/Os in parallel was created. The test was first executed 3 times with the Weighted Priority Queue (WPQ) scheduler. This provided a baseline to compare against.
+
+After this, the same test was executed with mClock scheduler and with different mClock profiles, i.e. *high_client_ops*, *balanced* and *high_recovery_ops*. The results collated for comparison. The test was executed 3 times for each mClock profile, and the average of those runs are reported in this study.
+
+## Establish Baseline Client Throughput (IOPS)
+
+For the tests, NVMe SSDs were used as the backing device for the OSDs. A 1:1 mapping of an OSD to the device was set up. But before the actual tests, the baseline throughput was established by running a 100% 4KiB random write ‘fio’ benchmark on a single RBD pool configured with the same OSDs to be used in the tests, and with a replication factor of 3. An average baseline client throughput of around 23K IOPS (91 MiB/s) @ 4KiB random writes was established. The bluestore throttle parameters namely `bluestore_throttle_bytes` and `bluestore_throttle_deferred_bytes` were determined to be 256 KiB using these [steps](https://docs.ceph.com/en/latest/rados/configuration/mclock-config-ref/#steps-to-manually-benchmark-an-osd-optional) (Note that ‘fio’ tool was used instead of OSD bench).
```

Comment:
Comma after "parameters"
```suggestion
For the tests, NVMe SSDs were used as the backing device for the OSDs. A 1:1 mapping of an OSD to the device was set up. But before the actual tests, the baseline throughput was established by running a 100% 4KiB random write ‘fio’ benchmark on a single RBD pool configured with the same OSDs to be used in the tests, and with a replication factor of 3. An average baseline client throughput of around 23K IOPS (91 MiB/s) @ 4KiB random writes was established. The bluestore throttle parameters, namely `bluestore_throttle_bytes` and `bluestore_throttle_deferred_bytes` were determined to be 256 KiB using these [steps](https://docs.ceph.com/en/latest/rados/configuration/mclock-config-ref/#steps-to-manually-benchmark-an-osd-optional) (Note that ‘fio’ tool was used instead of OSD bench).
```

Reply:
Done.